tree/30_generic_methods/condition_from_command.cf

```diff
@@ -64,12 +65,22 @@ bundle agent condition_from_command(condition_prefix, command, true_codes, false
       "true_list"     slist => splitstring("${true_codes}", "\s*,\s*", "256");
       "false_list"    slist => splitstring("${false_codes}", "\s*,\s*", "256");
 
+  classes:
+      "pass3" expression => "pass2";
+      "pass2" expression => "pass1";
+      "pass1" expression => "any";
+
   methods:
+    pass1.!pass2::
+      "remove_dry_run_${class_prefix}"  usebundle => push_dry_run_mode("false");
+    pass2.!pass3::
+      "restore_dry_run_${class_prefix}" usebundle => pop_dry_run_mode();
+
       "report"
-        usebundle  => _log_v3("Execute the test command ${command} to create ${condition_prefix}_{true,false}", "${condition_prefix}", "${old_class_prefix}", "${class_prefix}", @{args}),
-        ifvarclass => "${class_prefix}_reached";
+        usebundle  => _log_v3("Execute the test command ${command} to create ${condition_prefix}_{true,false}", "${condition_prefix}", "${old_class_prefix}", "${class_prefix}", @{args});
```

Comment:
shouldn't it be done `pass3::`?

Reply:
Does not really matter since the "action" part is over in pass1.
But at least for clarity we should put pass3.

Reply:
Commands are done after methods so I think it would fail